### PR TITLE
[CPDLP-3510] npq internal api revisit fallback for funding eligibility logic in npq

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

Ticket: https://dfedigital.atlassian.net.mcas.ms/browse/CPDLP-3510

### Changes proposed in this pull request

* Replace all the `Rails.application.config.npq_separation[:ecf_api_disabled]` flags to use `Feature.ecf_api_disabled?`
* Updated `Ecf::NpqProfileUpdater` spec to include one more test

### CODE REVIEW

**Review by cwrw — COMMENTED** [review]
Thanks @mooktakim is it ok if we only focus on swapping out the feature flag on the funding eligibility section? as there is another ticket to tackle those which will change their location?

**Review by leandroalemao — COMMENTED** [review]
All good past Abeer's comment.. also Is the logic behind `accepted_applications` and `with_targeted_delivery_funding_eligibility` the same as in ECF?

File: spec/lib/services/ecf/npq_profile_updater_spec.rb [hunk 1]
```diff
@@ -123,10 +123,14 @@
   end
 
   context "when ecf_api_disabled flag is toggled on" do
-    before { allow(Rails.application.config).to receive(:npq_separation).and_return({ ecf_api_disabled: true }) }
+    before { allow(Feature).to receive(:ecf_api_disabled?).and_return(true) }
 
-    it "returns nil" do
+    it ".call returns nil" do
```
[the remark this answers is not in the document]
Comment (mooktakim) [reply spec/lib/services/ecf/npq_profile_updater_spec.rb]: those changes reverted now

[the remark this answers is not in the document]
Comment (mooktakim) [reply spec/lib/services/ecf/npq_profile_updater_spec.rb]: it was because `call` was tested but `tsf_data_field_update` wasn't.

**Review by leandroalemao — APPROVED** [review]

**Review by cwrw — APPROVED** [review]

**Pull request merged**